File: TP1-PHP/Electronic.php
<?php

class Electronic {
    // Attribute declaration (private)
    private string $id_toko;
    private string $product_name;
    private string $category;
    private int $price;
    private string $image_file_name;

    // Constructor
    public function __construct(string $id_toko, string $product_name, string $category, int $price, string $image_file_name ="") {
        $this->id_toko = $id_toko;
        $this->product_name = $product_name;
        $this->category = $category;
        $this->price = $price;
        $this->image_file_name = $image_file_name;
    }

    // Getter and Setter for id_toko
    public function getId(): string {
        return $this->id_toko;
    }

    public function setId(string $id_toko): void {
        $this->id_toko = $id_toko;
    }

    // Getter and Setter for product_name
    public function getProductName(): string {
        return $this->product_name;
    }

    public function setProductName(string $product_name): void {
        $this->product_name = $product_name;
    }

    // Getter and Setter for category
    public function getCategory(): string {
        return $this->category;
    }

    public function setCategory(string $category): void {
        $this->category = $category;
    }

    // Getter and Setter for price
    public function getPrice(): int {
        return $this->price;
    }

    public function setPrice(int $price): void {
        $this->price = $price;
    }
    
    // Getter and Setter for Image
    public function getImageFileName(): string {
    return $this->image_file_name;
}

public function setImageFileName(string $fileName): void {
    $this->image_file_name = $fileName;
}
}

?>

File: TP1-Python/Electronic.py
class Electronic:

    #attribute declaration
    def __init__(self, id_toko: str, product_name: str, category: str, price: int):
        self.__id_toko= str(id_toko)
        self.__product_name = str(product_name)
        self.__category = str(category)
        self.__price = int(price)

    #getter and setter for id attribute 
    def getId(self)->str:
        return self.__id_toko
    
    def setId(self, id_toko: str)->None:
        self.__id_toko = str(id_toko)

    #getter for product nama attribute 
    def getProductName(self)->str:
        return self.__product_name
    
    def setProductName(self, product_name: str)->None:
        self.__product_name = str(product_name)
    
    #getter for category attribute 
    def getCategory(self)->str:
        return self.__category
    
    def setCategory(self, product_name: str)->None:
        self.__category = str(product_name)
    
    #getter for price attribute 
    def getPrice(self)->int:
        return self.__price
    
    def setPrice(self, price: int)->None:
        self.__price = int(price)

    #function to display product information
    def product_information(self):
        print(f"Kode: {self.getId()}") #display id
        print(f"Nama: {self.getProductName()}") #display product name
        print(f"Kategori: {self.getCategory()}") #display  product category
        print(f"Harga: Rp {self.getPrice():,}") #display product price
        print("-" * 30)
    

